webpack/components/Score.jsx

```diff
@@ -1,190 +1,229 @@
-    "3",
-    "4",
-    "",
-    "5",
-    "6",
-    "",
-    "7",
-    "",
-    "8"
-  ];
-  const currentBeatNums = [
-    "",
-    "1",
-    "",
-    "2",
-    "",
-    "3",
-    "",
-    "4",
-    "",
-    "5",
-    "",
-    "6",
-    "",
-    "7",
-    "",
-    "8"
-  ];
-  const nextBeatNums = [
-    "",
-    "1",
-    "",
-    "2",
-    "",
-    "3",
-    "4",
-    "",
-    "5",
-    "6",
-    "",
-    "7",
-    "",
-    "8"
-  ];
-  const prevBeats = prevBeatNums.map((num, idx) => (
-    <CellBeat beatText={num} key={`beatNum${idx}`} /> // eslint-disable-line react/no-array-index-key
-  ));
-  const currentBeats = currentBeatNums.map((num, idx) => (
-    <CellBeat beatText={num} key={`beatNum${idx}`} /> // eslint-disable-line react/no-array-index-key
-  ));
-  const nextBeats = nextBeatNums.map((num, idx) => (
-    <CellBeat beatText={num} key={`beatNum${idx}`} /> // eslint-disable-line react/no-array-index-key
-  ));
-  return (
-    <div className="score">
-      <div className="measure measure--prev">
-        <MeasureLabelContainer previous />
-        <div className="measure__grid-container">
-          <div className="measure__channel">{prevBeats}</div>
-          <div className="measure__channel measure__channel--large">
-            <div className="cell cell--1 cell--text" />
-            <div className="cell cell--1 cell--text" />
-            <div className="cell cell--1 cell--text" />
-            <div className="cell cell--1 cell--text" />
-            <div className="cell cell--1 cell--text">ke</div>
-            <div className="cell cell--1 cell--text">i</div>
-            <div className="cell cell--1 cell--text">ko</div>
-            <div className="cell cell--1 cell--text">o</div>
-            <div className="cell cell--1 cell--text">te</div>
-            <div className="cell cell--1 cell--text">n</div>
-            <div className="cell cell--1 cell--text">no</div>
-            <div className="cell cell--1 cell--text">o</div>
-            <div className="cell cell--1 cell--text" />
-            <div className="cell cell--1 cell--text" />
-          </div>
-          <div className="measure__channel">
-            <div className="cell cell--14">Tsuzuke</div>
-          </div>
-          <div className="measure__channel">
-            <div className="cell cell--2 cell--dance" />
-            <div className="cell cell--9 cell--dance">grid</div>
-            <div className="cell cell--3 cell--dance" />
-          </div>
-          <div className="measure__channel">
-            <div className="cell cell--3" />
-            <div className="cell cell--6 cell--dance">
-              Sits back on his knee (9)
+function createBeatsArray(grid) {
+  const beatsArray = [];
+  for (let i = 0; i < grid.length; i += 1) {
+    beatsArray[grid[i].start] = grid[i].text;
+  }
+  for (let i = 0; i < beatsArray.length; i += 1) {
+    if (typeof beatsArray[i] === "undefined") {
+      beatsArray[i] = "";
+    }
+  }
+  return beatsArray;
+}
```

Comment:
The second loop could be rewritten as
```jsx
  return [...Array(beatsArray.length).keys()].map(i => beatsArray[i] || "")
```
There might be a way to do it all in one loop.